tokio-tls/src/lib.rs

```diff
@@ -20,10 +21,19 @@
 //! built. Configuration of TLS parameters is still primarily done through the
 //! `native-tls` crate.
 
-use futures::{Async, Future, Poll};
-use native_tls::{Error, HandshakeError};
+use native_tls::{Error, HandshakeError, MidHandshakeTlsStream};
+use std::future::Future;
 use std::io::{self, Read, Write};
-use tokio_io::{try_nb, AsyncRead, AsyncWrite};
+use std::marker::Unpin;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio_io::{AsyncRead, AsyncWrite};
+
+#[derive(Debug)]
+struct AllowStd<S> {
+    inner: S,
+    context: *mut (),
```

Comment:
This `*mut ()` makes the whole thing non-Send, which breaks integration with `hyper-tls`, where `Connect` expects the Transport type to be `Send`.

```
83 | impl<T> Connect for HttpsConnector<T>
   |         ^^^^^^^ `*mut ()` cannot be sent between threads safely
   |
   = help: within `stream::MaybeHttpsStream<<T as hyper::client::connect::Connect>::Transport>`, the trait `std::marker::Send` is not implemented for `*mut ()`
   = note: required because it appears within the type `tokio_tls::AllowStd<<T as hyper::client::connect::Connect>::Transport>`
   = note: required because it appears within the type `schannel::tls_stream::TlsStream<tokio_tls::AllowStd<<T as hyper::client::connect::Connect>::Transport>>`
   = note: required because it appears within the type `native_tls::imp::TlsStream<tokio_tls::AllowStd<<T as hyper::client::connect::Connect>::Transport>>`
   = note: required because it appears within the type `native_tls::TlsStream<tokio_tls::AllowStd<<T as hyper::client::connect::Connect>::Transport>>`
   = note: required because it appears within the type `tokio_tls::TlsStream<<T as hyper::client::connect::Connect>::Transport>`
   = note: required because it appears within the type `stream::MaybeHttpsStream<<T as hyper::client::connect::Connect>::Transport>`
```

Reply:
This should have been fixed on master.

Reply:
@carllerche, thank you, I've seen it :)